Add GltfLoaderSettings

robtfm opened the pull request:
# Objective

when loading gltfs we may want to filter the results. in particular, i need to be able to exclude cameras. 

i can do this already by modifying the gltf after load and before spawning, but it seems like a useful general option.

## Solution

add `GltfLoaderSettings` struct with bool members:
- `load_cameras` : checked before processing camera nodes.
- `load_lights` : checked before processing light nodes
- `load_meshes` : checked before loading meshes, materials and morph weights

Existing code will work as before. Now you also have the option to restrict what parts of the gltf are loaded. For example, to load a gltf but exclude the cameras, replace a call to `asset_server.load("my.gltf")` with:
```rust
asset_server.load_with_settings(
    "my.gltf",
    |s: &mut GltfLoaderSettings| {
        s.load_cameras = false;
    }
);
```

alice-i-cecile reviewed (changes requested):
The implementation looks good, and this should definitely exist.

I'd like a bit more documentation on how to use this in practice: right now, it's quite hard to piece the usage together.

alice-i-cecile commented:
I think it would also make sense to add a `load_lights` field to `GltfLoaderSettings`, but that can be done in another PR :)

robtfm commented:
added lights, and meshes for completeness.

not sure what makes this a breaking change? anyway i added the struct comment as migration guide, let me know if you want something else.

alice-i-cecile commented on crates/bevy_gltf/src/loader.rs:
```diff
@@ -105,16 +105,34 @@ pub struct GltfLoader {
-/// Specifies optional settings for processing gltfs at load time
+/// Specifies optional settings for processing gltfs at load time. By default, all recognized contents of
+/// the gltf will be loaded.
+/// Example usage: to load a gltf but exclude the cameras, replace a call to `asset_server.load("my.gltf")` with
```
```suggestion
/// # Example
///
/// To load a gltf but exclude the cameras, replace a call to `asset_server.load("my.gltf")` with
```

alice-i-cecile commented on crates/bevy_gltf/src/loader.rs:
```diff
@@ -105,20 +105,53 @@ pub struct GltfLoader {
 
+/// Specifies optional settings for processing gltfs at load time. By default, all recognized contents of
+/// the gltf will be loaded.
+/// # Example
```
```suggestion
///
/// # Example
```

alice-i-cecile commented:
Really nice, thank you!

The associated type on a public trait impl changed, which is technically semver breaking: [line 144](https://github.com/bevyengine/bevy/pull/10804/files#diff-4bad6321ea999887884bc321768f82c3bbd5aaeaf1b2ae4dca6e871b330426ebR144).

alice-i-cecile commented:
@robftm missing import in the example :)

robtfm commented:
my pc is even worse at running doctests than i am at writing them ...
